### ISSUE TITLE
ConfusionMatrix classes: one label must be specified in y true. 

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#244]
When trying to passes class names to the ConfusionMatrix, I'm getting the following error:

```
/usr/local/lib/python3.6/site-packages/sklearn/metrics/classification.py:248: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if np.all([l not in y_true for l in labels]):
Traceback (most recent call last):
  File "rc.py", line 63, in <module>
    viz.score(X_test, y_test)
  File "/Users/benjamin/Repos/ddl/yellowbrick/yellowbrick/classifier/confusion_matrix.py", line 107, in score
    y, y_pred, labels=self.classes_, sample_weight=sample_weight
  File "/usr/local/lib/python3.6/site-packages/sklearn/metrics/classification.py", line 249, in confusion_matrix
    raise ValueError("At least one label specified must be in y_true")
ValueError: At least one label specified must be in y_true
```

This is do to the fact that the y_true vector is label encoded (e.g. ones and zeros) but that the labels are strings, e.g. "occupied" and "unoccupied". 

Here is the code to produce the error:

```python 
df = pd.read_csv('examples/data/occupancy/occupancy.csv')

features = ["temperature", "relative humidity", "light", "C02", "humidity"]
target = "occupancy"

X = df[features]
y = df[target]
classes = ["unoccupied", "occupied"]

X_train, X_test, y_train, y_test = tts(X, y, test_size=0.2)

viz = ConfusionMatrix(MultinomialNB(), classes=classes)
viz.fit(X_train, y_train)
viz.score(X_test, y_test)
viz.poof()
```

Note that:

```
viz = ConfusionMatrix(MultinomialNB())
```

Does not raise an exception.

**Comment by cjmorale** [comment]
@bbengfort Is anyone working on this bug yet? I was able to reproduce the error last night on my local environment and wanted to offer my help.

**Comment by ndanielsen** [comment]
@cjmorale I'm one of the maintainers of yb, you're welcome to take a stab at it! 💯

**Comment by cjmorale** [comment]
@ndanielsen How should the system handle examples like @bbengfort provided above? I was a little unsure what type of behavior the yb team is looking for in these situations.

**Comment by NealHumphrey** [comment]
@cjmorale I'm another maintainer and also authored the original ConfusionMatrix. 

@bbengfort The error raised is actually the one raised by the scikitlearn. If I remember correctly, if you passed the same data directly into the scikitlearn confusion matrix you'd get the same error:

`sklearn.metrics.confusion_matrix(y_true, y_pred, labels=classes)`

If you want to have label encoded data but have text-friendly labels, we need the mapping. 

Proposed solution:
- Add another optional parameter to the __init__ which is `label_encoder`
- If the class list and the data don't match, try passing the class list through the reverse label encoder ([`label_encoder.inverse_transform()`](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html)
- Pass the matching version into the sklearn.metrics.confusion_matrix
- Put the text labels onto the graph

Bonus: 
- let the label_encoder param also accept a dictionary mapping of `{'encoded_number':'text label', '1':'apples'}` in case people don't use label encoder.

**Comment by NealHumphrey** [comment]
Side note: the functionality currently provided by the use of 'classes' is slicing the ConfusionMatrix, the same way the sklearn.metrics.confusion_matrix works. In case you were wondering why there was even a classes argument to begin with.

**Comment by cjmorale** [comment]
@NealHumphrey Thanks for your reply. I plan on working on this over the next few days so stay tuned.

**Comment by bbengfort** [comment]
So I also have this issue in the `ROCAUC` visualizer; and the scoring metrics are more sensitive to what's being passed in as `y` (which is usually `y_test`) -- in the end, I used the classes only for labels and

**Comment by rebeccabilbro** [comment]
Hey there @cjmorale @bbengfort @NealHumphrey! 

Just doing a quick inventory of old issues; I see that @cjmorale's fix for this was merged in and the issue was closed, but then re-opened. I'd like to update the issue to reflect what is left to do. From [here](https://github.com/DistrictDataLabs/yellowbrick/pull/253) it looks like we just need to update the documentation to reflect the merged changes?

**Comment by bbengfort** [comment]
@rebeccabilbro yep, sorry about this, it should have been closed on June 8 with #253. For future reference, the fix to this is as follows:

```python
viz = ConfusionMatrix(
    MultinomialNB(), classes=classes, label_encoder={0: "unoccupied", 1: "occupied"}
)
```

Where the `label_encoder` argument can be a `sklearn.preprocessing.LabelEncoder` (or really anything with `inverse_transform` that does the mapping) or a `dict`. 

closed via ff10ca0
